redfish-core/lib/log_services.hpp

```diff
@@ -4773,6 +4801,614 @@ inline void getSystemHardwareIsolationLogService(
+ * replace the redfish processor id as "dcmN-cpuN" because redfish currently
+ * does not support chip module concept.
+ *
+ * @param[in] dbusObjPath - The D-Bus object path to return the object instance
+ *
+ * @return the object instance with it parent instance id if the given object
+ *         is a processor else the object instance alone.
+ */
+inline std::string
+    getIsolatedHwItemId(const sdbusplus::message::object_path& dbusObjPath)
+{
+    std::string isolatedHwItemId;
+
+    if ((dbusObjPath.filename().find("cpu") != std::string::npos) &&
+        (dbusObjPath.parent_path().filename().find("dcm") != std::string::npos))
+    {
+        isolatedHwItemId = std::string(dbusObjPath.parent_path().filename() +
+                                       "-" + dbusObjPath.filename());
+    }
+    else
+    {
+        isolatedHwItemId = dbusObjPath.filename();
+    }
+    return isolatedHwItemId;
+}
+
+/**
+ * @brief API used to get redfish uri of the given dbus object and fill into
+ *        "OriginOfCondition" property of LogEntry schema.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ *
+ * @return The redfish response with "OriginOfCondition" property of
+ *         LogEntry schema if success else return the error
+ */
+inline void getRedfishUriByDbusObjPath(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const sdbusplus::message::object_path& dbusObjPath,
+    const size_t entryJsonIdx)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, dbusObjPath,
+         entryJsonIdx](const boost::system::error_code ec,
+                       const dbus::utility::MapperGetObject& objType) {
+        if (ec || objType.empty())
+        {
+            BMCWEB_LOG_ERROR << "DBUS response error [" << ec.value() << " : "
+                             << ec.message()
+                             << "] when tried to get the RedfishURI of "
+                             << "isolated hareware: " << dbusObjPath.str;
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        RedfishUriListType::const_iterator redfishUriIt;
+        for (const auto& service : objType)
+        {
+            for (const auto& interface : service.second)
+            {
+                redfishUriIt = redfishUriList.find(interface);
+                if (redfishUriIt != redfishUriList.end())
+                {
+                    // Found the Redfish URI of the isolated hardware unit.
+                    break;
+                }
+            }
+            if (redfishUriIt != redfishUriList.end())
+            {
+                // No need to check in the next service interface list
+                break;
+            }
+        }
```

Comment:
L4865-L4882 is a logic to search `RedfishResourceCollectionUri` (i.e. std::string) from `objType`.
Can it be simpler if we do this?

```
RedfishResourceCollectionUri foundRedfishURI = findRedfishUri(objType);
if(foundRedfishURI.empty())
{
    // Not found
}
```
And
```
inline RedfishResourceCollectionUri findRedfishUri( const dbus::utility::MapperGetObject& objType )
{
        for (const auto& service : objType)
        {
            for (const auto& interface : service.second)
            {
                auto& redfishUriIt = redfishUriList.find(interface);
                if (redfishUriIt != redfishUriList.end())
                {
                    // Found the Redfish URI of the isolated hardware unit.
                    return redfishUriIt->second;
                }
            }
        }
           return RedfishResourceCollectionUri{};
}
```